harness/determined/_train_context.py

```diff
@@ -176,6 +176,9 @@ def set_stop_requested(self, stop_requested: bool) -> None:
         )
         self._stop_requested = stop_requested
 
+    def get_initial_batch(self) -> int:
```

Comment:
This function's existence needs more consideration.

We can only provide this information reliably when we have a lot of control over the checkpoint state (ie. not in the generic API).

Maybe it's ok to leave in the Trial context, since we do have that level of control, it almost seems like we could be a lot less magical about this by having the (very few) users who need this value do this:

```python
initial_batch = 0
last_checkpoint = generic_context.checkpointing.get_restart_checkpoint()  # this API doesn't exist yet
if last_checkpoint is not None:
    inital_batch = last_checkpoint.get_metadata("last_batch")  # this would be checkpoint metadata
                                                               # that is always saved by our
                                                               # TrialController
```

I say this is "less magical" because it makes the TrialContext a little less different than the generic api context.